modules/component/src/core/cd-aware.abstract.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  ChangeDetectorRef,
+  EmbeddedViewRef,
+  NgZone,
+  OnDestroy,
+  Type,
+} from '@angular/core';
+import {
+  getChangeDetectionHandler,
+  getGlobalThis,
+  getRequestAnimationFrameInAngular,
+  potentialObservableValue,
+  remainHigherOrder,
+} from './utils';
+import {
+  defer,
+  NextObserver,
+  Observable,
+  PartialObserver,
+  Subject,
+  Subscription,
+} from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { processCdAwareObservables } from './operators';
+
+export interface CoalescingConfig {
+  optimized: boolean;
+}
+
+// This abstract class holds all the shared logic for the push pipe and the let directive
+// responsible for change detection
+// If you extend this class you need to implement how the update of the rendered value happens.
+// Also custom behaviour is something you need to implement in the extending class
+export abstract class CdAware implements OnDestroy {
+  handleChangeDetection: <T>(component?: T) => void = getChangeDetectionHandler(
+    this.ngZone,
+    this.cdRef
+  );
+  protected requestAnimationFrameRef: (
+    cb: () => void
+  ) => number = getRequestAnimationFrameInAngular().bind(getGlobalThis());
+  protected subscription = new Subscription();
+  protected observablesSubject = new Subject<potentialObservableValue<any>>();
+  // We have to defer the setup of observables$ until subscription as getConfigurableBehaviour is defined in the
+  // extending class. So getConfigurableBehaviour is not available in the abstract layer
+  protected observables$ = defer(() =>
+    this.observablesSubject.pipe(
+      processCdAwareObservables(
+        this.getResetContextBehaviour(),
+        this.getUpdateContextBehaviour(),
+        this.getConfigurableBehaviour()
+      )
+    )
+  );
+
+  constructor(
+    protected readonly cdRef: ChangeDetectorRef,
+    protected readonly ngZone: NgZone
+  ) {}
+
+  work(): void {
+    this.handleChangeDetection(
+      // cast is needed to make is work for typescript.
```

Comment:
"make is work" should probably be "make it work"

Reply:
done